newIDE/app/src/UI/KeyboardShortcuts/index.js

```diff
@@ -68,24 +86,81 @@ export default class KeyboardShortcuts {
     return this._shiftPressed;
   }
 
+  shouldFollowAxis() {
+    return this._shiftPressed;
+  }
+
+  shouldNotSnapToGrid() {
+    return this._altPressed;
+  }
+
+  shouldResizeProportionally() {
+    return this._shiftPressed;
+  }
+
+  shouldScrollHorizontally() {
+    return this._altPressed;
+  }
+
+  shouldMoveView() {
+    return this._spacePressed || this._mouseMidButtonPressed;
+  }
+
+  shouldZoom() {
+    if (isMacLike()) {
+      return this._isControlOrCmdPressed();
+    } else {
+      if (
+        !this._isControlOrCmdPressed() &&
+        !this._altPressed &&
+        !this._shiftPressed
+      ) {
+        return true;
+      } else {
+        return false;
+      }
+    }
+  }
+
   _updateModifiersFromEvent = (evt: KeyboardEvent | DragEvent) => {
     this._metaPressed = evt.metaKey;
     this._altPressed = evt.altKey;
     this._ctrlPressed = evt.ctrlKey;
     this._shiftPressed = evt.shiftKey;
   };
 
+  _updateSpecialKeysStatus = (evt: KeyboardEvent, isDown: boolean) => {
+    if (evt.which === SPACE_KEY) {
+      this._spacePressed = isDown;
+    }
+  };
+
   _isControlOrCmdPressed = () => {
     // On macOS, meta key (Apple/Command key) acts as Control key on Windows/Linux.
     return this._metaPressed || this._ctrlPressed;
   };
 
+  onMouseDown = (evt: MouseEvent) => {
+    if (evt.button === MID_MOUSE_BUTTON) {
+      this._mouseMidButtonPressed = true;
+    } else {
+      this._mouseMidButtonPressed = false;
```

Comment:
hmm, is it normal to go back to false if the button pressed is not the mid?
(it means pressing mid, then left, would return mid to false)

Reply:
I kept the original behaviour that was in the DeprecatedKeyboardShortcut.
I think it makes sense since you usually press a single button at a time.